crates/zizmor/src/github_api.rs

```diff
@@ -190,89 +281,94 @@ impl Client {
                 .build()
                 .expect("couldn't build GitHub client?"),
         )
-        .with(Cache(HttpCache {
-            mode: CacheMode::Default,
-            manager: CACacheManager {
-                path: cache_dir.into(),
-                remove_opts: Default::default(),
-            },
-            options: HttpCacheOptions {
-                cache_options: Some(CacheOptions {
-                    // GitHub API requests made with an API token seem to
-                    // always have `Cache-Control: private`, so we need to
-                    // explicitly tell http-cache that our cache is not shared
-                    // in order for things to cache correctly.
-                    shared: false,
-                    ..Default::default()
-                }),
-                ..Default::default()
-            },
-        }))
+        .with(CacheLoggingMiddleware)
+        .with(ChainedCache(
+            Cache(HttpCache {
+                mode: CacheMode::Default,
+                manager: CACacheManager {
+                    path: cache_dir.into(),
+                    remove_opts: Default::default(),
+                },
+                options: http_cache_options.clone(),
+            }),
+            CacheType::File,
+        ))
+        .with(ChainedCache(
+            Cache(HttpCache {
+                mode: CacheMode::ForceCache,
+                manager: MokaManager::new(MokaCache::new(1000)),
+                options: http_cache_options,
+            }),
+            CacheType::Memory,
+        ))
```

Comment:
@Bo98 just making sure I understand the intended behavior here -- is the idea here that we'll hit the in-memory cache first, and then fall back to the file cache if the former misses? In other words, we'll end up priming the memory cache with the contents of the file cache, if it has a hit?

Reply:
Thanks, that's helpful!